molpipeline/estimators/chemprop/lightning_wrapper.py

```diff
@@ -0,0 +1,161 @@
+            break
+    else:
+        enable_progress_model_summary = False
+    default_values = {
+        "strategy": "auto",
+        "limit_predict_batches": 1.0,
+        "fast_dev_run": False,
+        "min_steps": None,
+        "accumulate_grad_batches": 1,
+        "use_distributed_sampler": True,
+        "devices": "auto",
+        "check_val_every_n_epoch": 1,
+        "enable_progress_bar": True,
+        "max_epochs": None,
+        "max_time": None,
+        "val_check_interval": 1.0,
+        "log_every_n_steps": 50,
+        "min_epochs": None,
+        "gradient_clip_algorithm": None,
+        "max_steps": -1,
+        "limit_val_batches": 1.0,
+        "gradient_clip_val": None,
+        "inference_mode": True,
+        "enable_model_summary": None,
+        "limit_test_batches": 1.0,
+        "reload_dataloaders_every_n_epochs": 0,
+        "deterministic": False,
+        "logger": None,
+        "overfit_batches": 0.0,
+        "precision": "32-true",
+        "benchmark": False,
+        "num_sanity_val_steps": 2,
+        "enable_checkpointing": None,
+        "limit_train_batches": 1.0,
+        "barebones": False,
+        "detect_anomaly": False,
+        "num_nodes": 1,
+    }
+    trainer_dict = {
+        "accelerator": get_device(trainer),
+        "strategy": "auto",  # trainer.strategy, # collides with accelerator
+        "devices": "auto",  # trainer._accelerator_connector._devices_flag does not really work
+        "num_nodes": trainer.num_nodes,
+        "precision": trainer.precision,
+        "logger": trainer.logger,
+        "callbacks": trainer.callbacks,  # type: ignore[attr-defined]
+        "fast_dev_run": trainer.fast_dev_run,  # type: ignore[attr-defined]
+        "max_epochs": trainer.max_epochs,
+        "min_epochs": trainer.min_epochs,
+        "max_steps": trainer.max_steps,
+        "min_steps": trainer.min_steps,
+        "max_time": max_time,
+        "limit_train_batches": trainer.limit_train_batches,
+        "limit_val_batches": trainer.limit_val_batches,
+        "limit_test_batches": trainer.limit_test_batches,
+        "limit_predict_batches": trainer.limit_predict_batches,
+        "overfit_batches": trainer.overfit_batches,  # type: ignore[attr-defined]
+        "val_check_interval": trainer.val_check_interval,
+        "check_val_every_n_epoch": trainer.check_val_every_n_epoch,
+        "num_sanity_val_steps": trainer.num_sanity_val_steps,
+        "log_every_n_steps": trainer.log_every_n_steps,  # type: ignore[attr-defined]
+        "enable_checkpointing": bool(trainer.checkpoint_callbacks),
+        "enable_progress_bar": get_enable_progress_bar(trainer),
+        "enable_model_summary": enable_progress_model_summary,
+        "accumulate_grad_batches": trainer.accumulate_grad_batches,
+        "gradient_clip_val": trainer.gradient_clip_val,
+        "gradient_clip_algorithm": trainer.gradient_clip_algorithm,
+        "deterministic": torch.are_deterministic_algorithms_enabled(),
```

Comment:
is this a requirement to set deterministic to true? Or could this also be triggered by setting deterministic to true?

Reply:
I checked the code and it seems that lightning just sets this variable. But I will add a unittest to confirm if this works.

Reply:
Added a [unit test](https://github.com/basf/MolPipeline/blob/952fceaba72d659d4cde5297d1bfb01a50804216/test_extras/test_chemprop/test_lightning_wrapper.py) and confirmed that it works.